lib/stormpath-sdk/auth/register_service_provider.rb

```diff
@@ -0,0 +1,46 @@
+module Stormpath
+  module Authentication
+    class RegisterServiceProvider
+      attr_reader :client, :identity_provider, :assertion_consumer_service_url,
+                  :entity_id, :name, :description, :name_id_format
+
+      def initialize(client, identity_provider, assertion_consumer_service_url, entity_id, options = {})
+        @client = client
+        @identity_provider = identity_provider
+        @assertion_consumer_service_url = assertion_consumer_service_url
+        @entity_id = entity_id
+        @name = options[:name] || nil
+        @description = options[:description] || nil
+        @name_id_format = options[:name_id_format] || nil
```

Comment:
we dont need so many instance variables we can just put them in one called options